src/main.ts

```diff
@@ -873,9 +873,6 @@ function generateInterfaceDeclaration(
     const name = maybeSnakeToCamel(fieldDesc.name, options);
     const isOptional = isOptionalProperty(fieldDesc, messageDesc.options, options);
     const type = toTypeName(ctx, messageDesc, fieldDesc, isOptional);
-    if (isOptional && !type.toString().includes("undefined")) {
```

Comment:
The definitive fix probably lays within ts-poet, I believe the toString method shouldn't have any side-effects

Reply:
Wow, I agree that is very strange that `toString` has side-effects. Also kinda funny this `console.warn` was just hanging out here :shrug: . Thanks for finding! I'll see if I can poke around at the ts-poet root cause at some point...